### DISCUSSION TITLE
[cURL for Windows Issue] Do not support Chinese?

### DISCUSSION THREAD

**Discussion by iamxiaojianzheng** [mistweaverco/kulala.nvim#164] [Q&A] [answered]
```http
POST https://jsonplaceholder.typicode.com/posts
Content-Type: application/json; charset=utf-8

{"title":"中文"}
```

Chinese cannot be displayed properly

```json
{
  "title": "����",
  "id": 101
}
```

And when the post request parameter contains Chinese, it will be garbled.

![image](https://github.com/user-attachments/assets/246d3a99-27d0-44b4-ac72-4dcf997bbe85)

**Comment by gorillamoe** [comment]
Seems to work for me, at least when using httpbin.org.

![image](https://github.com/user-attachments/assets/c003082f-767a-48dd-8684-b6ccea7897b4)

```http
POST https://httpbin.org/post HTTP/1.1
Content-Type: application/json; charset=utf-8

{"title":"中文"}
```

Server responds with:

```json
{
  "args": {},
  "data": "{\"title\":\"中文\"}",
  "files": {},
  "form": {},
  "headers": {
    "Accept": "*/*",
    "Content-Length": "18",
    "Content-Type": "application/json; charset=utf-8",
    "Host": "httpbin.org",
    "User-Agent": "kulala.nvim/3.2.2",
    "X-Amzn-Trace-Id": "Root=1-66c0aedb-4fef252c64c025b71e58a522"
  },
  "json": {
    "title": "中文"
  },
  "origin": "80.187.119.86",
  "url": "https://httpbin.org/post"
}
```

**Comment by gorillamoe** [comment]
Variables also seem to work fine.

```http
GET https://httpbin.org/get?foo=中文 HTTP/1.1
```

Server responds with:

```json
{
  "args": {
    "foo": "中文"
  },
  "headers": {
    "Accept": "*/*",
    "Host": "httpbin.org",
    "User-Agent": "kulala.nvim/3.2.2",
    "X-Amzn-Trace-Id": "Root=1-66c0afaa-7b981b37247eabb4728b0e13"
  },
  "origin": "80.187.119.86",
  "url": "https://httpbin.org/get?foo=中文"
}
```

**Comment by iamxiaojianzheng** [comment]
Maybe it has something to do with the shell?

I tried `gitbash`, `powershell`, `cmd`. 

The following three results are obtained.

gitbash
![image](https://github.com/user-attachments/assets/f0a98166-4501-4c93-b883-64ed84888b0a)

powershell
![image](https://github.com/user-attachments/assets/8330ac7c-487c-4f79-8dc5-5858a75d2228)

cmd
![image](https://github.com/user-attachments/assets/66a9ddf4-c3ef-46c8-b184-0328a4fe5ed6)

**Comment by iamxiaojianzheng** [comment]
When variables are used, all three results are consistent.

![image](https://github.com/user-attachments/assets/45f1cc25-7e98-4090-a97b-fe1317053a21)

**Comment by iamxiaojianzheng** [comment]
curl version info

```
curl 8.9.0 (x86_64-w64-mingw32) libcurl/8.9.0 Schannel zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libpsl/0.21.5 libssh2/1.11.0
Release-Date: 2024-07-24
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli HSTS HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM PSL SPNEGO SSL SSPI threadsafe UnixSockets zstd
```

**Comment by gorillamoe** [comment]
Maybe it is either the shell or the curl version? Can you try wezterm?

**Comment by gorillamoe** [comment]
Seems to be an issue with curl for windows? https://github.com/curl/curl/discussions/10446

You can always try the `:lua require("kulala").copy()` command which should try to copy a curl command to your clipboard.

There you might add some `-vvv` flags to debug what curl is doing.

**Comment by iamxiaojianzheng** [comment]
It's a compatibility issue with curl on Windows.

I found a solution. Is it possible to do this in a configuration way in kulala?

```bash
echo -en "$(curl -s -X "POST" --data "{""title"":""中文""}" -H "content-type:application/json; charset=utf-8" --http1.1 -A "kulala.nvim/3.2.2" "https://httpbin.org/post")"
```

**Comment by gorillamoe** [comment]
I have to think about a workaround for that, but have you tried this? https://user-images.githubusercontent.com/2276718/227071658-2c25848f-fd69-444a-a95b-0e77cc2f5437.png

**Comment by iamxiaojianzheng** [comment]
This method, I have tried. But it still can't be solved.

Only slightly effective in gitbash

![image](https://github.com/user-attachments/assets/0687f836-34ab-4adc-a840-6df4796727e9)

**Comment by iamxiaojianzheng** [comment]
> This method, I have tried. But it still can't be solved.
> 
> Only slightly effective in gitbash
> 
> ![image](https://private-user-images.githubusercontent.com/29248764/358877292-0687f836-34ab-4adc-a840-6df4796727e9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5NDQ0ODUsIm5iZiI6MTcyMzk0NDE4NSwicGF0aCI6Ii8yOTI0ODc2NC8zNTg4NzcyOTItMDY4N2Y4MzYtMzRhYi00YWRjLWE4NDAtNmRmNDc5NjcyN2U5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODE4VDAxMjMwNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJiMjlkZWIwZjQ1YmJhZGU5ZWY2NDY5ODI3NDljMTA5NDUzMmJlNWNhNzRmZWE4OTAzODY5YzAxNDIxNGRkMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6BNftfT8tWPLX-rr86xJUjq-l98uikKCKJIWis-ISFU)

Compared to before

![image](https://github.com/user-attachments/assets/f0a98166-4501-4c93-b883-64ed84888b0a)

**Comment by gorillamoe** [comment]
Someone seems to have managed it with cmder:

- https://jonathansoma.com/everything/setup/utf-8/
- https://cmder.app/

**Comment by iamxiaojianzheng** [comment]
> I have to think about a workaround for that, but have you tried this? https://user-images.githubusercontent.com/2276718/227071658-2c25848f-fd69-444a-a95b-0e77cc2f5437.png

When I enable this, I have no problem accessing this interface.

![image](https://github.com/user-attachments/assets/c86625f4-fa7a-4c7a-9ba4-420022b16c42)

But access to this interface, the result is still Unicode.

![image](https://github.com/user-attachments/assets/e52e4e0b-8081-4290-af9d-99dbda08588c)

**Comment by gorillamoe** [comment, marked as answer]
> Someone seems to have managed it with cmder:
> 
> * https://jonathansoma.com/everything/setup/utf-8/
> * https://cmder.app/

I tried this by myself on windows and it's working.
